src/index.ts

```diff
@@ -21,70 +37,59 @@ export class TransactPluginTemplate extends AbstractTransactPlugin {
      * @param context The TransactContext of the transaction being performed
      */
     register(context: TransactContext): void {
-        // Optional - Retrieve the translation function from the UI if it exists
-        let t
-        if (context.ui) {
-            t = context.ui.getTranslate()
-        }
-
-        // Register any desired beforeSign hooks
+        // Register any desired afterBroadcast hooks
         context.addHook(
-            TransactHookTypes.beforeSign,
-            async (request, context): Promise<TransactHookResponseType> => {
-                // If this plugin is interacting with the UI, throw an error since this is an undefined function
-                if (context.ui) {
-                    throw new Error(
-                        // Translate the error message against the given key or use the default value as English
-                        t('beforesign', {
-                            default: 'undefined beforeSign hook called from plugin template',
+            TransactHookTypes.afterBroadcast,
+            (request, context): Promise<TransactHookResponseType> => {
+                setTimeout(async () => {
+                    this.log('Checking transaction finality')
+                    waitForFinality(request.getRawTransaction(), context)
```

Comment:
The `getRawTransaction` here isn't going to use the right transaction. I was just testing this out and the ID that was being queried was different than the transaction I submitted. 

I'm not sure we can get the real transaction out using the `request` object 🤔

Reply:
I'm going to change the `afterSign` and `afterBroadcast` hooks to pass in a `resolved` (ResolvedSigningRequest) value instead of `request` (SigningRequest).

Reply:
Alright, `@wharfkit/session` version `0.5.0` is released, and changes things to make this work. 

No longer is a `request` the first parameter of the hook, but instead `result` (which is a `TransactResult`), which has a bunch more information about what's happening.

You can see the modifications I made to the explorer plugin here: 

https://github.com/wharfkit/transact-plugin-explorerlink/commit/d392b1d088ff362f6374c41278441e19703b5244#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80L21-R20

You can also see where I'm getting the transaction ID from here:

https://github.com/wharfkit/transact-plugin-explorerlink/blob/da619afaa670a18947a185c6d92dc7f45ab756ad/src/index.ts#L75

Reply:
This comment applies to both the finality callback and checker plugins, they'll both need to be updated.

Reply:
Not sure how I missed these comments 😅 but I'm on it!

Reply:
Github notifications, lol... 🤦 

I honestly forgot I left them too. You asked a few times over the months and I said I'd get around to it... completely forgetting I even left these. So you're definitely not alone. We got a lotta stuff going on 😅

Reply:
Yeah lol, every time I asked you to take a look, I was looking at the other PR 🤦